File: game-server/src/main/java/work/fking/pangya/game/packet/handler/UserProfileRequestPacketHandler.java
package work.fking.pangya.game.packet.handler;

import io.netty.buffer.ByteBuf;
import work.fking.pangya.game.GameServer;
import work.fking.pangya.game.player.Player;
import work.fking.pangya.game.net.ClientGamePacketHandler;
import work.fking.pangya.game.packet.outbound.UserStatisticsReplies;

public class UserProfileRequestPacketHandler implements ClientGamePacketHandler {

    @Override
    public void handle(GameServer server, Player player, ByteBuf packet) {
        var userId = packet.readIntLE();
        var type = packet.readByte();

        var channel = player.channel();
        if (type == 5) {
            channel.write(UserStatisticsReplies.username(type, userId));
            channel.write(UserStatisticsReplies.character(userId));
            channel.write(UserStatisticsReplies.equipment(type, player));
        }
        channel.write(UserStatisticsReplies.userStatistic(type, userId));
        channel.write(UserStatisticsReplies.courseStatistic(type, userId));
        channel.write(UserStatisticsReplies.trophies(type, userId));
        channel.write(UserStatisticsReplies.ack(true, type, userId));
        channel.flush();
    }
}

File: game-server/src/main/kotlin/work/fking/pangya/game/packet/handler/UserProfileRequestPacketHandler.kt
package work.fking.pangya.game.packet.handler

import io.netty.buffer.ByteBuf
import work.fking.pangya.game.GameServer
import work.fking.pangya.game.net.ClientPacketHandler
import work.fking.pangya.game.packet.outbound.UserStatisticsReplies
import work.fking.pangya.game.player.Player

class UserProfileRequestPacketHandler : ClientPacketHandler {

    override fun handle(server: GameServer, player: Player, packet: ByteBuf) {
        val playerUid = packet.readIntLE()
        val type = packet.readByte().toInt()

        if (type == 5) {
            player.write(UserStatisticsReplies.username(type, player))
            player.write(UserStatisticsReplies.character(player))
            player.write(UserStatisticsReplies.equipment(type, player))
        }
        player.write(UserStatisticsReplies.userStatistic(type, playerUid, player.statistics))
        player.write(UserStatisticsReplies.courseStatistic(type, playerUid))
        player.write(UserStatisticsReplies.trophies(type, playerUid))
        player.write(UserStatisticsReplies.ack(true, type, playerUid))
        player.flush()
    }
}
